handlers.js

```diff
@@ -4,87 +4,113 @@ module.exports = self => {
-            const ns = cur._id;
-            const resources = self.formatPieces(cur.pieces);
-            acc[ns] = resources;
-            return acc;
-          }, {});
-
-          if (!isEqual(i18nextResources, i18nStaticResources)) {
-            modified = true;
-            // eslint-disable-next-line no-console
-            console.log(`Add missing pieces in i18n-static module for ${locale}...`);
-
-            for (const [ namespace, resources ] of Object.entries(i18nextResources)) {
-              for (const [ key, value ] of Object.entries(resources || {})) {
-                const [ title, pluralType ] = key.split('_');
-                const valueToCheck = plurals[pluralType] || 'valueSingular';
-                const props = {
-                  title,
-                  namespace
-                };
-
-                const existingPiece = await self.find(req, props).toObject();
-
-                if (!existingPiece) {
-                  await self.insert(req, {
-                    ...props,
-                    [valueToCheck]: value
-                  });
-                } else if (!existingPiece[valueToCheck]) {
-                  const newPiece = {
-                    ...existingPiece,
-                    [valueToCheck]: value
+        return await self.apos.lock.withLock('i18n-static-lock', async () => {
+          let modified = false;
+
+          const i18nextNamespaces = Object.keys(self.apos.i18n.namespaces)
+            .filter(ns => !self.options.excludeNamespaces.includes(ns));
+
+          const plurals = {
+            plural: 'valuePlural',
+            zero: 'valueZero',
+            two: 'valuePluralTwo',
+            few: 'valuePluralFew',
+            many: 'valuePluralMany'
+          };
+
+          for (const locale of self.apos.i18n.i18next.options.languages) {
+            const req = self.apos.task.getReq({
+              locale,
+              mode: 'draft'
+            });
+            const i18nextResources = i18nextNamespaces.reduce((acc, cur) => {
+              const resources = self.apos.i18n.i18next.getResourceBundle(locale, cur);
+              return {
+                ...acc,
+                ...(resources && { [cur]: resources })
+              };
+            },
+            {});
+
+            let cachedI18nStaticResources = await self.apos.cache.get(locale, 'i18n-static');
+            if (!cachedI18nStaticResources) {
+              cachedI18nStaticResources = await self.findPiecesAndGroupByNamespace(`${locale}:draft`);
+              await self.apos.cache.set(req.locale, 'i18n-static', cachedI18nStaticResources);
+            }
+            const i18nStaticResources = cachedI18nStaticResources.reduce((acc, cur) => {
+              const ns = cur._id;
+              const resources = self.formatPieces(cur.pieces);
+              acc[ns] = resources;
+              return acc;
+            }, {});
+
+            if (!isEqual(i18nextResources, i18nStaticResources)) {
```

Comment:
At this point we're just looking to make sure the keys present in the JSON all exist in the cache, right? I think this code might also be checking that the values are the same, which wouldn't make sense because it would force this logic to run on every startup as soon as at least one piece has been edited to have a translation that differs from the original JSON.

Reply:
If a value is different because a key was edited in another instance, should we test for equality? 
What would be your solution if you just want to test keys? Browse every key in both objects?